### ISSUE TITLE
[tmva][sofie] Keras parser: Conv2D padding='same' ignores dilation_rate, producing wrong padding

### ISSUE THREAD

**Issue by harz05** [root-project/root#21684]
### Check duplicate issues.

- [ ] Checked for duplicates

### Description

In `bindings/pyroot/pythonizations/python/ROOT/_pythonization/_tmva/_sofie/_parser/_keras/layers/conv.py`, the `padding='same'` calculation does not account for `dilation_rate`. The effective kernel size for a dilated convolution is `dilation * (kernel - 1) + 1`, but the code uses the raw kernel size directly:

```
# lines 55-56 — dilation_rate is extracted at line 36 but never used here
padding_height = max((outputHeight - 1) * fAttrStrides[0] + fAttrKernelShape[0] - inputHeight, 0)
padding_width  = max((outputWidth  - 1) * fAttrStrides[1] + fAttrKernelShape[1] - inputWidth,  0)
```

For `dilation=2, kernel=3, input=8, stride=1`:
- SOFIE computes total padding = 2
- Correct total padding = 4 (verified against Keras output shape)
- With half the required padding, the generated inference code operates on tensors with wrong spatial dimensions

### Reproducer

Step 1 — verify Keras ground truth: `padding='same'` with `dilation_rate=2` should preserve spatial dimensions:

```python
import keras
model = keras.Sequential([
    keras.Input(shape=(8, 8, 1)),
    keras.layers.Conv2D(1, kernel_size=3, padding='same', dilation_rate=2)
])
print('input shape: ', model.input_shape)   # (None, 8, 8, 1)
print('output shape:', model.output_shape)  # (None, 8, 8, 1)  <- same padding must preserve this
```

Step 2 — show what SOFIE actually computes vs the correct value:

```python
import math

inputHeight    = 8
fAttrStrides   = [1, 1]
fAttrKernelShape = [3, 3]
fAttrDilations = [2, 2]

outputHeight = math.ceil(inputHeight / fAttrStrides[0])

buggy   = max((outputHeight - 1) * fAttrStrides[0] + fAttrKernelShape[0] - inputHeight, 0)
eff_k   = fAttrDilations[0] * (fAttrKernelShape[0] - 1) + 1
correct = max((outputHeight - 1) * fAttrStrides[0] + eff_k - inputHeight, 0)

print(f'SOFIE computes:  {buggy}')   # 2
print(f'correct padding: {correct}') # 4
```

<img width="967" height="600" alt="Image" src="https://github.com/user-attachments/assets/24c0b144-2fb9-4f60-86c0-87d75422629f" />

### ROOT version

6.39.01 (built from source)

### Installation method

Built from source with `-Dtmva-sofie=ON -Dtmva-pymva=ON -Dbuiltin_protobuf=ON`

### Operating system

Linux (Ubuntu 22.04)

### Additional context

Only affects Conv2D with `padding='same'` and `dilation_rate > 1`. The padding='valid' path is unaffected since it does not compute explicit padding values.
